Section padding settings

tyleralsbury opened the pull request:
**Why are these changes introduced?**

Fixes #1000.

**What approach did you take?**

Added the settings to the sections that are mentioned in the Figmas attached to the issue.

I also added them to the main product page section and the main cart items section.  There may be other places that need this, too.

**Note:** I spoke to @Oliviammarcello and she mentioned that we are setting the default vertical padding between sections to `0`, so we should test with that as well and make sure there aren't any sections that will be too squished when vertical spacing is set to `0`.  This is how I found that it needed to be added to the product page and cart items.  There may be more that need the same treatment.

**Demo links**

- [Store](https://os2-demo.myshopify.com/?_ab=0&_fd=0&_sc=1&preview_theme_id=127010701334)
- [Editor](https://os2-demo.myshopify.com/admin/themes/127010701334/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

kmeleta commented on sections/rich-text.liquid:
```diff
@@ -4,8 +4,22 @@
+
+  @media screen and (min-width: 750px) {
+    .section-{{ section.id }}-padding {
+      padding-top: {{ section.settings.desktop_padding_top }}px;
+      padding-bottom: {{ section.settings.desktop_padding_bottom }}px;
```
These are being overridden by existing rich text padding styles.

kmeleta commented on sections/collection-list.liquid:
```diff
@@ -1,12 +1,25 @@
-{{ 'component-card.css' | asset_url | stylesheet_tag }}
+{%- style -%}
+  .section-{{ section.id }}-padding {
+    padding-top: {{ section.settings.mobile_padding_top }}px;
+    padding-bottom: {{ section.settings.mobile_padding_bottom }}px;
```
Mobile paddings being overridden by existing styles

kmeleta commented on sections/multicolumn.liquid:
```diff
@@ -2,9 +2,23 @@
 
+{%- style -%}
+  .section-{{ section.id }}-padding {
+    padding-top: {{ section.settings.mobile_padding_top }}px;
```
Same with these.

kmeleta reviewed (commented):
Did a quick first pass. Found a couple sections where we still need to remove some section-specific padding styles.

tyleralsbury commented:
> I realized I missed this in my mockups, but since the video section isn't full width it should have the section padding sliders as well

Should it only affect the section when it's not full width or regardless of "full width" being checked?

tyleralsbury commented:
Adjusted all points.  I've adjusted the padding to affect the secondary background for now, to ensure the behaviour works well until we make updates to the colour schemes.

tyleralsbury commented:
@Oliviammarcello Is this the final copy?  I would like to request translations.

tyleralsbury commented:
> > @Oliviammarcello Is this the final copy? I would like to request translations.
> 
> Just saw Melissa's note about removing all mobile settings. We should update the settings and copy to [look like this](https://screenshot.click/21-10-qo6qx-mx3s6.png)
> 
> cc @katycobb

Sounds good. Will get on that.

tyleralsbury commented:
> Email signup, rich text, email signup: Screenshot
> Top: 40px
> Bottom: 52px

> Featured collection, collection list, blog posts, multicolumn, collage, video, contact form:
> Top and bottom: 36px

Added new defaults

kmeleta commented on sections/rich-text.liquid:
```diff
@@ -4,8 +4,35 @@
+  assign section_mobile_padding_top = section.settings.padding_top
+  assign section_mobile_padding_bottom = section.settings.padding_bottom
+
+  if section.settings.padding_top >= 28 or section.settings.padding_top <= 80
```
These conditionals don't seem like they're working. I'm seeing negative paddings and 10px being subtracted from the value below 28px
<img src="https://screenshot.click/14-41-uqtsf-w1u6c.png" width="500" />

I think you might want an `and` instead of an `or`

kmeleta commented on sections/collage.liquid:
```diff
@@ -2,9 +2,36 @@
 {{ 'component-card.css' | asset_url | stylesheet_tag }}
 {{ 'component-price.css' | asset_url | stylesheet_tag }}
 
+{% liquid
+  assign section_mobile_padding_top = section.settings.padding_top
+  assign section_mobile_padding_bottom = section.settings.padding_bottom
+
+  if section.settings.padding_top >= 28 or section.settings.padding_top <= 80
+    assign section_mobile_padding_top = section.settings.padding_top | minus: 10
+  endif
+
+  if section.settings.padding_bottom >= 28 or section.settings.padding_bottom <= 80
+    assign section_mobile_padding_bottom = section.settings.padding_bottom | minus: 10
+  endif
+%}
+
+{%- style -%}
+  .section-{{ section.id }}-padding {
+    padding-top: {{ section_mobile_padding_top }}px;
+    padding-bottom: {{ section_mobile_padding_bottom }}px;
+  }
+
+  @media screen and (min-width: 750px) {
+    .section-{{ section.id }}-padding {
+      padding-top: {{ section.settings.padding_top }}px;
+      padding-bottom: {{ section.settings.padding_bottom }}px;
+    }
+  }
+{%- endstyle -%}
```
Any reason we couldn't make all this a snippet to be re-used in each section? Or are you trying to avoid the overuse of snippets? I feel like only having to make updates/fixes in one place would be cleaner and easier than a big find/replace.

tyleralsbury replied:
Bonus tip, by the way - you can do multi-line find/replace, so when the code is 100% identical between the different sections, you can actually get rid of whole big blocks like that.

tyleralsbury replied:
For now we've removed all of the math.  I will be making a follow-up PR that will explore using some fancy math to create a smoother curve.

Oliviammarcello reviewed (approved):
- [x] Add section padding to custom liquid section. 

Other than that LGTM 🚢

tyleralsbury commented:
> * [x]  Add section padding to custom liquid section.
> 
> Other than that LGTM 🚢

Done :)

kmeleta reviewed (approved):
The section paddings all look like they're doing what they should so I'm happy to approve, but I did find a couple instances of phantom paddings from internal section elements if we were interested to tackling those here. 

Overall I think our solution to vertical padding/margin works very well 👍  Much more sane than and predictable than all the conditional magic stuff. 

Video looks like the title container is still taking up space when no title is provided https://screenshot.click/17-45-9c7p1-zttlj.png

Grid could probably benefit from a margin-top: 0 but I also wouldn't say I'm fully aware of the implications of adding that. https://screenshot.click/17-46-cpyya-db3nl.png

tyleralsbury commented:
> Video looks like the title container is still taking up space when no title is provided https://screenshot.click/17-45-9c7p1-zttlj.png

Fixed that one.  Good catch.  I fixed a few more titles, too.

We really need to revisit titles in general and simplify them.  They just keep getting more and more complex and I think the complexity mostly came from art direction from the original Dawn.  With it being a lot more flexible and providing settings, I think we can simplify the component as a whole and make all titles behave more naturally with less code.

> Grid could probably benefit from a margin-top: 0 but I also wouldn't say I'm fully aware of the implications of adding that. https://screenshot.click/17-46-cpyya-db3nl.png

Good point.  For now I left it as-is.  We can take a look at it and decide what to do - it might just be "Not cute".

kmeleta commented:
> We really need to revisit titles in general and simplify them

Totally agree. They've really been a bit of an annoyance the whole way with these new settings.

Oliviammarcello commented:
> > * [x]   Add section padding to custom liquid section.
> > 
> > Other than that LGTM 🚢
> 
> Done :)

Sorry the "Section padding" label is missing!

KaichenWang commented on assets/collage.css:
```diff
@@ -1,4 +1,5 @@
 .collage-wrapper-title {
```
Could probably be standardized and simply use something like `.title` in the markup instead of one-off `.collage-wrapper-title`. Can look into this with the other ["title" issues](https://github.com/Shopify/dawn/pull/1006#issuecomment-996892693) you mentioned.

tyleralsbury replied:
Created an issue a while back for it: https://github.com/Shopify/dawn/issues/286

I want to get rid all of the wrapper classes except maybe the one that handles titles with link.  They are a real mess.

KaichenWang reviewed (approved):
Looks great! 

### Some other sections may also benefit from this setting

#### Collection template - Product grid section

![Screen_Shot_2021-12-17_at_2_01_34_PM](https://user-images.githubusercontent.com/3710548/146597734-418c45aa-39e8-4070-b8b8-511bd17aacaf.jpg)

#### Blog template - Blog posts section
![Screen_Shot_2021-12-17_at_2_09_56_PM](https://user-images.githubusercontent.com/3710548/146597824-f9246cae-622e-4cee-a8cd-7abc0ed691d0.jpg)

#### Search template - Search results section

![Screen_Shot_2021-12-17_at_2_09_07_PM](https://user-images.githubusercontent.com/3710548/146597988-0a3a7c5a-dbc3-4a5b-8c37-919c67f9efb9.jpg)

### Not a section, but could still use some padding

#### Blog post template - Blog post section
![Screen_Shot_2021-12-17_at_2_04_40_PM](https://user-images.githubusercontent.com/3710548/146598112-13840b39-6738-411e-bf91-97537e02687e.jpg)
